tools/test_full_sync.py

```diff
@@ -16,7 +19,41 @@
 from chia.cmds.init_funcs import chia_init
 
 
-async def run_sync_test(file: Path, db_version=2) -> None:
+exit_with_failure = False
```

Comment:
after spending 5 minutes googling, I couldn't figure out how to pass a pointer/reference to an `int` in python, that's all. I suppose I could pass in an empty dict and lookup a key in it. It seems a bit excessive.

Reply:
No, you can't have pointers, just references to objects.  Well, I made some pointer'ish things once but they are quite foul and never to be used beyond the experimentation of creating them.  Anyways, why do you need that?  If this were an attribute on `ExitOnError` then it could be set in `.emit()` as `self.error_seen = True`.  Instead of `logger.addHandler(ExitOnError())` you could `exit_on_error_handler = ExitOnError()` before adding the handler.  And finally, at the point of checking it you would `if exit_on_error_handler.exit_seen:`.

If something fancier were needed then you could pass an `asyncio.Task()` to `ExitOnError()` and it could `.cancel()` that task.